Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1826,8 +1825,11 @@ LayoutUnit RenderFlexibleBox::computeCrossSizeForChildUsingContainerCrossSize(co
         // Let's compute the definite size value for the flex item (value that we can resolve without running layout).
         auto isHorizontal = isHorizontalFlow();
         auto size = isHorizontal ? style().height() : style().width();
-        ASSERT(size.isFixed());
         auto definiteValue = LayoutUnit { size.value() };
+        if (std::optional<LayoutUnit> percentageBasedSize; size.isPercent() && (percentageBasedSize = containingBlockLogicalHeightForContent(AvailableLogicalHeightType::IncludeMarginBorderPadding)))
+            definiteValue = percentageBasedSize.value();
+        else
+            definiteValue = size.value();
```

Comment:
Looks like the `else` is not needed?

Reply:
Ah yes you are right. I think what I was supposed to do was change definiteValue so that it was only declared and not defined above. The code then would have looked something like this: 

        auto definiteValue;
        if (std::optional<LayoutUnit> percentageBasedSize; size.isPercent() && (percentageBasedSize = containingBlockLogicalHeightForContent(AvailableLogicalHeightType::IncludeMarginBorderPadding)))
            definiteValue = percentageBasedSize.value();
        else
            definiteValue = size.value();

        Do you think one version is better than the other or have a preference? One benefit of this would be that definiteValue only gets written once but I think either will work.

Reply:
I think Cathie is right and looks like you can avoid the percentageBasedSize temp var as well.

Reply:
Ok I will change that logic. So do you think the check for size.isPercent() is sufficient for that if clause then? what happens if containingBlockLogicalHeightForContent is empty/null?

Reply:
In cases like this usually value_or is used.

Reply:
Ok the changes have been made. I also realized containingBlockLogicalHeightForContent was the wrong method to use here. That was setting the cross size based off of the total available logical height, but we need it based off the % here. I think availableLogicalHeightForPercentageComputation is the right method to use here. I'll also be updating the test case to make sure it covers this.

Reply:
Patch looks good but I wonder in what cases availableLogicalHeightForPercentageComputation can return null? And also if null is returned whether using 0_ul is the right thing to do.

Reply:
Looking this again, I think min-width/height and max-width/height could use the same treatment (support percentage), but I think it is enough to leave a FIXME for that.